Use simpler memory fencing for Perfgraph

friedmud opened the pull request:
Hopefully will be easier for threading tools to see what's going on

BIG NOTE: I think that the previous implementation of this was _correct_ and more performant... but it was definitely more complicated and ripe for failure if someone modified it incorrectly.

closes #19443

friedmud committed:
Use simpler memory fencing that will hopefully be easier for threading tools to see closes #19443

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/951254/) on 292f0c5 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20248/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/951264/) on 292f0c5 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>c068d4</th>
      <th colspan=4>#20248 292f0c</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.31%</td>
      <td>82.31%</td>
      <td>-</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>73477</td>
      <td>73477</td>
      <td>-</td>
      <td>2</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15795</td>
      <td>15795</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20248/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20248/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20248/coverage/xfem)

</details>



This comment will be updated on new commits.

lindsayad commented:
Threading diagnostic tools are definitely happier now. There are now only two races reported by `ThreadSanitizer` when run on `simple_diffusion`
```
WARNING: ThreadSanitizer: data race (pid=3551029)
  Write of size 8 at 0x7f68f5b8fee8 by main thread:
    #0 PerfGraph::setLiveTimeLimit(double) /home/lindad/projects4/moose/framework/build/header_symlinks/PerfGraph.h:150:67 (libmoose-devel.so.0+0x10879d6)
    #1 CommonOutputAction::act() /home/lindad/projects4/moose/framework/src/actions/CommonOutputAction.C:238:17 (libmoose-devel.so.0+0x10879d6)
    #2 Action::timedAct() /home/lindad/projects4/moose/framework/src/actions/Action.C:93:3 (libmoose-devel.so.0+0x1062e2d)
    #3 ActionWarehouse::executeActionsWithAction(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /home/lindad/projects4/moose/framework/src/actions/ActionWarehouse.C:384:19 (libmoose-devel.so.0+0x106a98a)
    #4 ActionWarehouse::executeAllActions() /home/lindad/projects4/moose/framework/src/actions/ActionWarehouse.C:344:5 (libmoose-devel.so.0+0x1069f79)
    #5 MooseApp::runInputFile() /home/lindad/projects4/moose/framework/src/base/MooseApp.C:1027:21 (libmoose-devel.so.0+0x1e767ea)
    #6 MooseApp::run() /home/lindad/projects4/moose/framework/src/base/MooseApp.C:1456:5 (libmoose-devel.so.0+0x1e7bf56)
    #7 main /home/lindad/projects4/moose/test/src/main.C:33:8 (moose_test-devel+0x4bd41d)

  Previous read of size 8 at 0x7f68f5b8fee8 by thread T1 (mutexes: write M2013):
    #0 std::chrono::duration<double, std::ratio<1l, 1l> >::duration<double, void>(double const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/chrono:522:27 (libmoose-devel.so.0+0x1ed418d)
    #1 PerfGraphLivePrint::start() /home/lindad/projects4/moose/framework/src/utils/PerfGraphLivePrint.C:321:9 (libmoose-devel.so.0+0x1ed418d)
    #2 PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0::operator()() const /home/lindad/projects4/moose/framework/src/utils/PerfGraph.C:54:61 (libmoose-devel.so.0+0x1f14ef2)
    #3 void std::__invoke_impl<void, PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>(std::__invoke_other, PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (libmoose-devel.so.0+0x1f14ef2)
    #4 std::__invoke_result<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>::type std::__invoke<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>(PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (libmoose-devel.so.0+0x1f14ef2)
    #5 void std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (libmoose-devel.so.0+0x1f14ef2)
    #6 std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (libmoose-devel.so.0+0x1f14ef2)
    #7 std::thread::_State_impl<std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (libmoose-devel.so.0+0x1f14ef2)
    #8 <null> <null> (libstdc++.so.6+0xda693)

  Location is heap block of size 401160 at 0x7f68f5b2e000 allocated by main thread:
    #0 operator new(unsigned long) <null> (moose_test-devel+0x4bae96)
    #1 std::_MakeUniq<PerfGraph>::__single_object std::make_unique<PerfGraph, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:30 (libmoose-devel.so.0+0x1eb6986)
    #2 RestartableData<PerfGraph>::RestartableData<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, void*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /home/lindad/projects4/moose/framework/build/header_symlinks/RestartableData.h:97:18 (libmoose-devel.so.0+0x1eb6986)
    #3 std::_MakeUniq<RestartableData<PerfGraph> >::__single_object std::make_unique<RestartableData<PerfGraph>, char const (&) [11], MooseApp*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(char const (&) [11], MooseApp*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (libmoose-devel.so.0+0x1e9e623)
    #4 MooseApp::createRecoverablePerfGraph() /home/lindad/projects4/moose/framework/src/base/MooseApp.C:2712:7 (libmoose-devel.so.0+0x1e6c377)
    #5 MooseApp::MooseApp(InputParameters) /home/lindad/projects4/moose/framework/src/base/MooseApp.C:338:17 (libmoose-devel.so.0+0x1e681f0)
    #6 MooseTestApp::MooseTestApp(InputParameters const&) /home/lindad/projects4/moose/test/src/base/MooseTestApp.C:60:66 (libmoose_test-devel.so.0+0x98b72d)
    #7 void __gnu_cxx::new_allocator<MooseTestApp>::construct<MooseTestApp, InputParameters const&>(MooseTestApp*, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/ext/new_allocator.h:162:23 (libmoose_test-devel.so.0+0x98e3f5)
    #8 void std::allocator_traits<std::allocator<MooseTestApp> >::construct<MooseTestApp, InputParameters const&>(std::allocator<MooseTestApp>&, MooseTestApp*, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/alloc_traits.h:512:8 (libmoose_test-devel.so.0+0x98e3f5)
    #9 std::_Sp_counted_ptr_inplace<MooseTestApp, std::allocator<MooseTestApp>, (__gnu_cxx::_Lock_policy)2>::_Sp_counted_ptr_inplace<InputParameters const&>(std::allocator<MooseTestApp>, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:519:4 (libmoose_test-devel.so.0+0x98e3f5)
    #10 std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<MooseTestApp, std::allocator<MooseTestApp>, InputParameters const&>(MooseTestApp*&, std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:651:6 (libmoose_test-devel.so.0+0x98e3f5)
    #11 std::__shared_ptr<MooseTestApp, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<std::allocator<MooseTestApp>, InputParameters const&>(std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:1342:14 (libmoose_test-devel.so.0+0x98e3f5)
    #12 std::shared_ptr<MooseTestApp>::shared_ptr<std::allocator<MooseTestApp>, InputParameters const&>(std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:409:4 (libmoose_test-devel.so.0+0x98e3f5)
    #13 std::shared_ptr<MooseTestApp> std::allocate_shared<MooseTestApp, std::allocator<MooseTestApp>, InputParameters const&>(std::allocator<MooseTestApp> const&, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:862:14 (libmoose_test-devel.so.0+0x98e3f5)
    #14 std::shared_ptr<MooseTestApp> std::make_shared<MooseTestApp, InputParameters const&>(InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:878:14 (libmoose_test-devel.so.0+0x98e3f5)
    #15 std::shared_ptr<MooseApp> AppFactory::buildApp<MooseTestApp>(InputParameters const&) /home/lindad/projects4/moose/framework/build/header_symlinks/AppFactory.h:132:12 (libmoose_test-devel.so.0+0x98e3f5)
    #16 AppFactory::createShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters, int) /home/lindad/projects4/moose/framework/src/base/AppFactory.C:87:10 (libmoose-devel.so.0+0x1e50548)
    #17 AppFactory::createAppShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, char**, int) /home/lindad/projects4/moose/framework/src/base/AppFactory.C:55:33 (libmoose-devel.so.0+0x1e4fe3d)
    #18 main /home/lindad/projects4/moose/test/src/main.C:30:35 (moose_test-devel+0x4bd3f9)

  Mutex M2013 (0x7f68f5b8fe90) created at:
    #0 pthread_mutex_lock <null> (moose_test-devel+0x447c06)
    #1 __gthread_mutex_lock(pthread_mutex_t*) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/x86_64-linux-gnu/c++/11/bits/gthr-default.h:749:12 (libmoose-devel.so.0+0x1ed415d)
    #2 std::mutex::lock() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_mutex.h:100:17 (libmoose-devel.so.0+0x1ed415d)
    #3 std::unique_lock<std::mutex>::lock() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_lock.h:139:17 (libmoose-devel.so.0+0x1ed415d)
    #4 std::unique_lock<std::mutex>::unique_lock(std::mutex&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_lock.h:69:2 (libmoose-devel.so.0+0x1ed415d)
    #5 PerfGraphLivePrint::start() /home/lindad/projects4/moose/framework/src/utils/PerfGraphLivePrint.C:304:34 (libmoose-devel.so.0+0x1ed415d)
    #6 PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0::operator()() const /home/lindad/projects4/moose/framework/src/utils/PerfGraph.C:54:61 (libmoose-devel.so.0+0x1f14ef2)
    #7 void std::__invoke_impl<void, PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>(std::__invoke_other, PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (libmoose-devel.so.0+0x1f14ef2)
    #8 std::__invoke_result<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>::type std::__invoke<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>(PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (libmoose-devel.so.0+0x1f14ef2)
    #9 void std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (libmoose-devel.so.0+0x1f14ef2)
    #10 std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (libmoose-devel.so.0+0x1f14ef2)
    #11 std::thread::_State_impl<std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (libmoose-devel.so.0+0x1f14ef2)
    #12 <null> <null> (libstdc++.so.6+0xda693)

  Thread T1 (tid=3551031, running) created by main thread at:
    #0 pthread_create <null> (moose_test-devel+0x429e5b)
    #1 std::thread::_M_start_thread(std::unique_ptr<std::thread::_State, std::default_delete<std::thread::_State> >, void (*)()) <null> (libstdc++.so.6+0xda969)
    #2 std::_MakeUniq<PerfGraph>::__single_object std::make_unique<PerfGraph, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (libmoose-devel.so.0+0x1eb69c1)
    #3 RestartableData<PerfGraph>::RestartableData<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, void*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /home/lindad/projects4/moose/framework/build/header_symlinks/RestartableData.h:97:18 (libmoose-devel.so.0+0x1eb69c1)
    #4 std::_MakeUniq<RestartableData<PerfGraph> >::__single_object std::make_unique<RestartableData<PerfGraph>, char const (&) [11], MooseApp*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(char const (&) [11], MooseApp*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (libmoose-devel.so.0+0x1e9e623)
    #5 MooseApp::createRecoverablePerfGraph() /home/lindad/projects4/moose/framework/src/base/MooseApp.C:2712:7 (libmoose-devel.so.0+0x1e6c377)
    #6 MooseApp::MooseApp(InputParameters) /home/lindad/projects4/moose/framework/src/base/MooseApp.C:338:17 (libmoose-devel.so.0+0x1e681f0)
    #7 MooseTestApp::MooseTestApp(InputParameters const&) /home/lindad/projects4/moose/test/src/base/MooseTestApp.C:60:66 (libmoose_test-devel.so.0+0x98b72d)
    #8 void __gnu_cxx::new_allocator<MooseTestApp>::construct<MooseTestApp, InputParameters const&>(MooseTestApp*, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/ext/new_allocator.h:162:23 (libmoose_test-devel.so.0+0x98e3f5)
    #9 void std::allocator_traits<std::allocator<MooseTestApp> >::construct<MooseTestApp, InputParameters const&>(std::allocator<MooseTestApp>&, MooseTestApp*, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/alloc_traits.h:512:8 (libmoose_test-devel.so.0+0x98e3f5)
    #10 std::_Sp_counted_ptr_inplace<MooseTestApp, std::allocator<MooseTestApp>, (__gnu_cxx::_Lock_policy)2>::_Sp_counted_ptr_inplace<InputParameters const&>(std::allocator<MooseTestApp>, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:519:4 (libmoose_test-devel.so.0+0x98e3f5)
    #11 std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<MooseTestApp, std::allocator<MooseTestApp>, InputParameters const&>(MooseTestApp*&, std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:651:6 (libmoose_test-devel.so.0+0x98e3f5)
    #12 std::__shared_ptr<MooseTestApp, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<std::allocator<MooseTestApp>, InputParameters const&>(std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:1342:14 (libmoose_test-devel.so.0+0x98e3f5)
    #13 std::shared_ptr<MooseTestApp>::shared_ptr<std::allocator<MooseTestApp>, InputParameters const&>(std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:409:4 (libmoose_test-devel.so.0+0x98e3f5)
    #14 std::shared_ptr<MooseTestApp> std::allocate_shared<MooseTestApp, std::allocator<MooseTestApp>, InputParameters const&>(std::allocator<MooseTestApp> const&, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:862:14 (libmoose_test-devel.so.0+0x98e3f5)
    #15 std::shared_ptr<MooseTestApp> std::make_shared<MooseTestApp, InputParameters const&>(InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:878:14 (libmoose_test-devel.so.0+0x98e3f5)
    #16 std::shared_ptr<MooseApp> AppFactory::buildApp<MooseTestApp>(InputParameters const&) /home/lindad/projects4/moose/framework/build/header_symlinks/AppFactory.h:132:12 (libmoose_test-devel.so.0+0x98e3f5)
    #17 AppFactory::createShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters, int) /home/lindad/projects4/moose/framework/src/base/AppFactory.C:87:10 (libmoose-devel.so.0+0x1e50548)
    #18 AppFactory::createAppShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, char**, int) /home/lindad/projects4/moose/framework/src/base/AppFactory.C:55:33 (libmoose-devel.so.0+0x1e4fe3d)
    #19 main /home/lindad/projects4/moose/test/src/main.C:30:35 (moose_test-devel+0x4bd3f9)

SUMMARY: ThreadSanitizer: data race /home/lindad/projects4/moose/framework/build/header_symlinks/PerfGraph.h:150:67 in PerfGraph::setLiveTimeLimit(double)
==================
==================
WARNING: ThreadSanitizer: data race (pid=3551029)
  Write of size 4 at 0x7f68f5b8fef0 by main thread:
    #0 PerfGraph::setLiveMemoryLimit(unsigned int) /home/lindad/projects4/moose/framework/build/header_symlinks/PerfGraph.h:155:75 (libmoose-devel.so.0+0x1087aa7)
    #1 CommonOutputAction::act() /home/lindad/projects4/moose/framework/src/actions/CommonOutputAction.C:239:17 (libmoose-devel.so.0+0x1087aa7)
    #2 Action::timedAct() /home/lindad/projects4/moose/framework/src/actions/Action.C:93:3 (libmoose-devel.so.0+0x1062e2d)
    #3 ActionWarehouse::executeActionsWithAction(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /home/lindad/projects4/moose/framework/src/actions/ActionWarehouse.C:384:19 (libmoose-devel.so.0+0x106a98a)
    #4 ActionWarehouse::executeAllActions() /home/lindad/projects4/moose/framework/src/actions/ActionWarehouse.C:344:5 (libmoose-devel.so.0+0x1069f79)
    #5 MooseApp::runInputFile() /home/lindad/projects4/moose/framework/src/base/MooseApp.C:1027:21 (libmoose-devel.so.0+0x1e767ea)
    #6 MooseApp::run() /home/lindad/projects4/moose/framework/src/base/MooseApp.C:1456:5 (libmoose-devel.so.0+0x1e7bf56)
    #7 main /home/lindad/projects4/moose/test/src/main.C:33:8 (moose_test-devel+0x4bd41d)

  Previous read of size 4 at 0x7f68f5b8fef0 by thread T1 (mutexes: write M2013):
    #0 PerfGraphLivePrint::iterateThroughExecutionList() /home/lindad/projects4/moose/framework/src/utils/PerfGraphLivePrint.C:281:62 (libmoose-devel.so.0+0x1ed3b4c)
    #1 PerfGraphLivePrint::start() /home/lindad/projects4/moose/framework/src/utils/PerfGraphLivePrint.C:378:5 (libmoose-devel.so.0+0x1ed4443)
    #2 PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0::operator()() const /home/lindad/projects4/moose/framework/src/utils/PerfGraph.C:54:61 (libmoose-devel.so.0+0x1f14ef2)
    #3 void std::__invoke_impl<void, PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>(std::__invoke_other, PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (libmoose-devel.so.0+0x1f14ef2)
    #4 std::__invoke_result<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>::type std::__invoke<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>(PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (libmoose-devel.so.0+0x1f14ef2)
    #5 void std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (libmoose-devel.so.0+0x1f14ef2)
    #6 std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (libmoose-devel.so.0+0x1f14ef2)
    #7 std::thread::_State_impl<std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (libmoose-devel.so.0+0x1f14ef2)
    #8 <null> <null> (libstdc++.so.6+0xda693)

  Location is heap block of size 401160 at 0x7f68f5b2e000 allocated by main thread:
    #0 operator new(unsigned long) <null> (moose_test-devel+0x4bae96)
    #1 std::_MakeUniq<PerfGraph>::__single_object std::make_unique<PerfGraph, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:30 (libmoose-devel.so.0+0x1eb6986)
    #2 RestartableData<PerfGraph>::RestartableData<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, void*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /home/lindad/projects4/moose/framework/build/header_symlinks/RestartableData.h:97:18 (libmoose-devel.so.0+0x1eb6986)
    #3 std::_MakeUniq<RestartableData<PerfGraph> >::__single_object std::make_unique<RestartableData<PerfGraph>, char const (&) [11], MooseApp*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(char const (&) [11], MooseApp*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (libmoose-devel.so.0+0x1e9e623)
    #4 MooseApp::createRecoverablePerfGraph() /home/lindad/projects4/moose/framework/src/base/MooseApp.C:2712:7 (libmoose-devel.so.0+0x1e6c377)
    #5 MooseApp::MooseApp(InputParameters) /home/lindad/projects4/moose/framework/src/base/MooseApp.C:338:17 (libmoose-devel.so.0+0x1e681f0)
    #6 MooseTestApp::MooseTestApp(InputParameters const&) /home/lindad/projects4/moose/test/src/base/MooseTestApp.C:60:66 (libmoose_test-devel.so.0+0x98b72d)
    #7 void __gnu_cxx::new_allocator<MooseTestApp>::construct<MooseTestApp, InputParameters const&>(MooseTestApp*, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/ext/new_allocator.h:162:23 (libmoose_test-devel.so.0+0x98e3f5)
    #8 void std::allocator_traits<std::allocator<MooseTestApp> >::construct<MooseTestApp, InputParameters const&>(std::allocator<MooseTestApp>&, MooseTestApp*, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/alloc_traits.h:512:8 (libmoose_test-devel.so.0+0x98e3f5)
    #9 std::_Sp_counted_ptr_inplace<MooseTestApp, std::allocator<MooseTestApp>, (__gnu_cxx::_Lock_policy)2>::_Sp_counted_ptr_inplace<InputParameters const&>(std::allocator<MooseTestApp>, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:519:4 (libmoose_test-devel.so.0+0x98e3f5)
    #10 std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<MooseTestApp, std::allocator<MooseTestApp>, InputParameters const&>(MooseTestApp*&, std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:651:6 (libmoose_test-devel.so.0+0x98e3f5)
    #11 std::__shared_ptr<MooseTestApp, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<std::allocator<MooseTestApp>, InputParameters const&>(std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:1342:14 (libmoose_test-devel.so.0+0x98e3f5)
    #12 std::shared_ptr<MooseTestApp>::shared_ptr<std::allocator<MooseTestApp>, InputParameters const&>(std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:409:4 (libmoose_test-devel.so.0+0x98e3f5)
    #13 std::shared_ptr<MooseTestApp> std::allocate_shared<MooseTestApp, std::allocator<MooseTestApp>, InputParameters const&>(std::allocator<MooseTestApp> const&, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:862:14 (libmoose_test-devel.so.0+0x98e3f5)
    #14 std::shared_ptr<MooseTestApp> std::make_shared<MooseTestApp, InputParameters const&>(InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:878:14 (libmoose_test-devel.so.0+0x98e3f5)
    #15 std::shared_ptr<MooseApp> AppFactory::buildApp<MooseTestApp>(InputParameters const&) /home/lindad/projects4/moose/framework/build/header_symlinks/AppFactory.h:132:12 (libmoose_test-devel.so.0+0x98e3f5)
    #16 AppFactory::createShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters, int) /home/lindad/projects4/moose/framework/src/base/AppFactory.C:87:10 (libmoose-devel.so.0+0x1e50548)
    #17 AppFactory::createAppShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, char**, int) /home/lindad/projects4/moose/framework/src/base/AppFactory.C:55:33 (libmoose-devel.so.0+0x1e4fe3d)
    #18 main /home/lindad/projects4/moose/test/src/main.C:30:35 (moose_test-devel+0x4bd3f9)

  Mutex M2013 (0x7f68f5b8fe90) created at:
    #0 pthread_mutex_lock <null> (moose_test-devel+0x447c06)
    #1 __gthread_mutex_lock(pthread_mutex_t*) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/x86_64-linux-gnu/c++/11/bits/gthr-default.h:749:12 (libmoose-devel.so.0+0x1ed415d)
    #2 std::mutex::lock() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_mutex.h:100:17 (libmoose-devel.so.0+0x1ed415d)
    #3 std::unique_lock<std::mutex>::lock() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_lock.h:139:17 (libmoose-devel.so.0+0x1ed415d)
    #4 std::unique_lock<std::mutex>::unique_lock(std::mutex&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_lock.h:69:2 (libmoose-devel.so.0+0x1ed415d)
    #5 PerfGraphLivePrint::start() /home/lindad/projects4/moose/framework/src/utils/PerfGraphLivePrint.C:304:34 (libmoose-devel.so.0+0x1ed415d)
    #6 PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0::operator()() const /home/lindad/projects4/moose/framework/src/utils/PerfGraph.C:54:61 (libmoose-devel.so.0+0x1f14ef2)
    #7 void std::__invoke_impl<void, PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>(std::__invoke_other, PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (libmoose-devel.so.0+0x1f14ef2)
    #8 std::__invoke_result<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>::type std::__invoke<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0>(PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (libmoose-devel.so.0+0x1f14ef2)
    #9 void std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (libmoose-devel.so.0+0x1f14ef2)
    #10 std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (libmoose-devel.so.0+0x1f14ef2)
    #11 std::thread::_State_impl<std::thread::_Invoker<std::tuple<PerfGraph::PerfGraph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MooseApp&, bool, bool)::$_0> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (libmoose-devel.so.0+0x1f14ef2)
    #12 <null> <null> (libstdc++.so.6+0xda693)

  Thread T1 (tid=3551031, running) created by main thread at:
    #0 pthread_create <null> (moose_test-devel+0x429e5b)
    #1 std::thread::_M_start_thread(std::unique_ptr<std::thread::_State, std::default_delete<std::thread::_State> >, void (*)()) <null> (libstdc++.so.6+0xda969)
    #2 std::_MakeUniq<PerfGraph>::__single_object std::make_unique<PerfGraph, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (libmoose-devel.so.0+0x1eb69c1)
    #3 RestartableData<PerfGraph>::RestartableData<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, void*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /home/lindad/projects4/moose/framework/build/header_symlinks/RestartableData.h:97:18 (libmoose-devel.so.0+0x1eb69c1)
    #4 std::_MakeUniq<RestartableData<PerfGraph> >::__single_object std::make_unique<RestartableData<PerfGraph>, char const (&) [11], MooseApp*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, MooseApp&, bool const&, bool>(char const (&) [11], MooseApp*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, MooseApp&, bool const&, bool&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (libmoose-devel.so.0+0x1e9e623)
    #5 MooseApp::createRecoverablePerfGraph() /home/lindad/projects4/moose/framework/src/base/MooseApp.C:2712:7 (libmoose-devel.so.0+0x1e6c377)
    #6 MooseApp::MooseApp(InputParameters) /home/lindad/projects4/moose/framework/src/base/MooseApp.C:338:17 (libmoose-devel.so.0+0x1e681f0)
    #7 MooseTestApp::MooseTestApp(InputParameters const&) /home/lindad/projects4/moose/test/src/base/MooseTestApp.C:60:66 (libmoose_test-devel.so.0+0x98b72d)
    #8 void __gnu_cxx::new_allocator<MooseTestApp>::construct<MooseTestApp, InputParameters const&>(MooseTestApp*, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/ext/new_allocator.h:162:23 (libmoose_test-devel.so.0+0x98e3f5)
    #9 void std::allocator_traits<std::allocator<MooseTestApp> >::construct<MooseTestApp, InputParameters const&>(std::allocator<MooseTestApp>&, MooseTestApp*, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/alloc_traits.h:512:8 (libmoose_test-devel.so.0+0x98e3f5)
    #10 std::_Sp_counted_ptr_inplace<MooseTestApp, std::allocator<MooseTestApp>, (__gnu_cxx::_Lock_policy)2>::_Sp_counted_ptr_inplace<InputParameters const&>(std::allocator<MooseTestApp>, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:519:4 (libmoose_test-devel.so.0+0x98e3f5)
    #11 std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<MooseTestApp, std::allocator<MooseTestApp>, InputParameters const&>(MooseTestApp*&, std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:651:6 (libmoose_test-devel.so.0+0x98e3f5)
    #12 std::__shared_ptr<MooseTestApp, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<std::allocator<MooseTestApp>, InputParameters const&>(std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:1342:14 (libmoose_test-devel.so.0+0x98e3f5)
    #13 std::shared_ptr<MooseTestApp>::shared_ptr<std::allocator<MooseTestApp>, InputParameters const&>(std::_Sp_alloc_shared_tag<std::allocator<MooseTestApp> >, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:409:4 (libmoose_test-devel.so.0+0x98e3f5)
    #14 std::shared_ptr<MooseTestApp> std::allocate_shared<MooseTestApp, std::allocator<MooseTestApp>, InputParameters const&>(std::allocator<MooseTestApp> const&, InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:862:14 (libmoose_test-devel.so.0+0x98e3f5)
    #15 std::shared_ptr<MooseTestApp> std::make_shared<MooseTestApp, InputParameters const&>(InputParameters const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr.h:878:14 (libmoose_test-devel.so.0+0x98e3f5)
    #16 std::shared_ptr<MooseApp> AppFactory::buildApp<MooseTestApp>(InputParameters const&) /home/lindad/projects4/moose/framework/build/header_symlinks/AppFactory.h:132:12 (libmoose_test-devel.so.0+0x98e3f5)
    #17 AppFactory::createShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters, int) /home/lindad/projects4/moose/framework/src/base/AppFactory.C:87:10 (libmoose-devel.so.0+0x1e50548)
    #18 AppFactory::createAppShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, char**, int) /home/lindad/projects4/moose/framework/src/base/AppFactory.C:55:33 (libmoose-devel.so.0+0x1e4fe3d)
    #19 main /home/lindad/projects4/moose/test/src/main.C:30:35 (moose_test-devel+0x4bd3f9)
```

lindsayad committed:
Make time and memory limit atomic to avoid possible torn reads/writes

We can make these operations relaxed as I don't think we really care
about ordering

lindsayad commented:
Adding lindsayad/moose@8159163118 on top of what you've don here makes `simple_diffusion` report 100% clean

friedmud commented:
Thanks for checking this @lindsayad ... let me take a closer look at your fix here...
